File: app/src/main/java/zebrostudio/wallr100/ui/buypro/BuyProPresenterImpl.java
package zebrostudio.wallr100.ui.buypro;


import javax.inject.Inject;

public class BuyProPresenterImpl implements BuyProContract.BuyProPresenter {

    private BuyProContract.BuyProView mBuyProView;

    @Inject
    BuyProPresenterImpl(){

    }

    @Override
    public void bindView(BuyProContract.BuyProView view) {
        mBuyProView = view;
    }

    @Override
    public void unbindView() {
        mBuyProView = null;
    }

    @Override
    public void onBuyProClick() {

    }

    @Override
    public void onRestorePurchaseClick() {

    }
}

File: app/src/main/java/zebrostudio/wallr100/ui/buypro/BuyProPresenterImpl.kt
package zebrostudio.wallr100.ui.buypro

import zebrostudio.wallr100.data.DataRepository

class BuyProPresenterImpl(private var dataRepository: DataRepository) : BuyProContract.BuyProPresenter {

  private var buyProView: BuyProContract.BuyProView? = null

  override fun attachView(view: BuyProContract.BuyProView) {
    buyProView = view
    dataRepository.initPurchaseHelper()
  }

  override fun detachView() {
    dataRepository.disposePurchaseHelper()
    buyProView = null
  }

  override fun notifyBuyProClicked() {

  }

  override fun notifyRestoreProClicked() {

  }

}